auslib/admin/views/required_signoffs.py

```diff
@@ -0,0 +1,272 @@
+        return super(ProductRequiredSignoffsView, self)._post(form, transaction, changed_by)
+
+
+class ProductRequiredSignoffsHistoryAPIView(RequiredSignoffsHistoryAPIView):
+
+    def __init__(self):
+        super(ProductRequiredSignoffsHistoryAPIView, self).__init__(dbo.productRequiredSignoffs, ["product", "channel", "role"])
+
+    def get(self):
+        form = ProductRequiredSignoffHistoryForm(request.args)
+        return super(ProductRequiredSignoffsHistoryAPIView, self).get(form)
+
+
+class ProductRequiredSignoffsScheduledChangesView(ScheduledChangesView):
+    def __init__(self):
+        super(ProductRequiredSignoffsScheduledChangesView, self).__init__("product_req_signoffs", dbo.productRequiredSignoffs)
+
+    @requirelogin
+    def _post(self, transaction, changed_by):
+        change_type = request.json.get("change_type")
+
+        if change_type == "update":
+            form = ScheduledChangeExistingProductRequiredSignoffForm()
+        elif change_type == "insert":
+            form = ScheduledChangeNewProductRequiredSignoffForm()
+        elif change_type == "delete":
+            form = ScheduledChangeDeleteProductRequiredSignoffForm()
+        else:
+            return Response(status=400, response="Invalid or missing change_type")
+
+        return super(ProductRequiredSignoffsScheduledChangesView, self)._post(form, transaction, changed_by)
+
+
+class ProductRequiredSignoffScheduledChangeView(ScheduledChangeView):
+    def __init__(self):
+        super(ProductRequiredSignoffScheduledChangeView, self).__init__("product_req_signoffs", dbo.productRequiredSignoffs)
+
+    @requirelogin
+    def _post(self, sc_id, transaction, changed_by):
+        if request.json and request.json.get("data_version"):
+            form = EditScheduledChangeExistingProductRequiredSignoffForm()
+        else:
+            form = EditScheduledChangeNewProductRequiredSignoffForm()
+
+        return super(ProductRequiredSignoffScheduledChangeView, self)._post(sc_id, form, transaction, changed_by)
+
+    @requirelogin
+    def _delete(self, sc_id, transaction, changed_by):
+        return super(ProductRequiredSignoffScheduledChangeView, self)._delete(sc_id, transaction, changed_by)
+
+
+class EnactProductRequiredSignoffScheduledChangeView(EnactScheduledChangeView):
+    def __init__(self):
+        super(EnactProductRequiredSignoffScheduledChangeView, self).__init__("product_req_signoffs", dbo.productRequiredSignoffs)
+
+    @requirelogin
+    def _post(self, sc_id, transaction, changed_by):
+        return super(EnactProductRequiredSignoffScheduledChangeView, self)._post(sc_id, transaction, changed_by)
+
+
+class ProductRequiredSignoffScheduledChangeSignoffsView(SignoffsView):
+    def __init__(self):
+        super(ProductRequiredSignoffScheduledChangeSignoffsView, self).__init__("product_req_signoffs", dbo.productRequiredSignoffs)
+
+    @requirelogin
+    def _post(self, sc_id, transaction, changed_by):
```

Comment:
I think you're probably right that I could move requirelogin to the base class - I'll give it a try.